### PR TITLE
#562: get dags to run query refactoring

### LINKED ISSUE #562: Refactor DagInstanceRepository.getDagsToRun
Refactor DagInstanceRepository.getDagsToRun:
- Sort dag runs in db 
- Get only number of dag runs limited by size in db

### PR DESCRIPTION
**Pull request by jozefbakus**
Closes #562 

Generated query:
Screenshot 2021-12-21 at 10 01 39

To copy:
```
select *
from (
    select min("id") as minDagInstanceId
    from "dag_instance"
    where (("status" in ('InQueue', 'Running')) and ("workflow_id" in (?, ?, ?))) and (not ("workflow_id" in (?, ?, ?)))
    group by "workflow_id" order by min("id") limit 3
) subQuery, "dag_instance" di
where di."id" = subQuery.minDagInstanceId;
```

### CODE REVIEW

File: src/main/scala/za/co/absa/hyperdrive/trigger/persistance/DagInstanceRepository.scala [hunk 1]
```diff
@@ -75,20 +75,23 @@ class DagInstanceRepositoryImpl @Inject()(val dbProvider: DatabaseProvider) exte
...
-        .result
-    )
-
-    prefilteredResult.map(di =>
-      di.groupBy(_.workflowId).flatMap(_._2.sortBy(_.id).take(1)).take(size).toSeq
-    )
+    val dagIdsQuery = dagInstanceTable
+      .filter(_.status inSet DagInstanceStatuses.nonFinalStatuses)
+      .filter(_.workflowId inSetBind assignedWorkflowIds)
+      .filterNot(_.id inSet runningIds)
```
Comment (kevinwallimann) [inline src/main/scala/za/co/absa/hyperdrive/trigger/persistance/DagInstanceRepository.scala]: Is this simplification ok? In the original query, all dags from the same running workflow would be excluded

Comment (jozefbakus) [reply src/main/scala/za/co/absa/hyperdrive/trigger/persistance/DagInstanceRepository.scala]: Yes, you are right. Fixed

File: src/main/scala/za/co/absa/hyperdrive/trigger/persistance/DagInstanceRepository.scala [hunk 2]
```diff
@@ -91,6 +91,7 @@ class DagInstanceRepositoryImpl @Inject()(val dbProvider: DatabaseProvider) exte
       dag
     }
 
+    dagsToRunQuery.result.statements.foreach(println)
```
Comment (kevinwallimann) [inline src/main/scala/za/co/absa/hyperdrive/trigger/persistance/DagInstanceRepository.scala]: Did you forget to remove that?

Comment (jozefbakus) [reply src/main/scala/za/co/absa/hyperdrive/trigger/persistance/DagInstanceRepository.scala]: Removed

**Review by kevinwallimann — APPROVED** [review]

**Pull request merged**